File: data structures and algorithms/python/04_015.py
from abc import ABC, abstractmethod

class NewSystem(ABC):
    @abstractmethod
    def process_data(self):
        pass

class OldSystem:
    def process_legacy_data(self):
        # old logic
        pass

class SystemAdapter(NewSystem):
    def __init__(self, old_system):
        self.old_system = old_system
    
    def process_data(self):
        self.old_system.process_legacy_data()

File: data structures and algorithms/cs/04_015.cs
public interface INewSystem
{
   void ProcessData();
}

public class OldSystem
{
   public void ProcessLegacyData() { /* old logic */ }
}

public class SystemAdapter : INewSystem
{
   private OldSystem oldSystem;

   public SystemAdapter(OldSystem oldSystem)
   {
      this.oldSystem = oldSystem;
   }

   public void ProcessData()
   {
      oldSystem.ProcessLegacyData();
   }
}
